### PR TITLE
Switch to argparse for argument parsing

### PR DESCRIPTION
**Pull request by DanNixon**
As suggested in #43, switch to the ```argparse``` module for argument parsing.

### CODE REVIEW

File: src/codemod.py [hunk 1]
```diff
@@ -802,74 +737,111 @@ def _terminal_restore_color():
...
-    opts, remaining_args = getopt.gnu_getopt(
-        sys.argv[1:], 'md:',
-        ['start=', 'end=', 'extensions=', 'exclude_paths=',
-         'include-extensionless', 'editor=', 'count', 'test'])
-  except getopt.error:
-    raise _UsageException()
-  opts = dict(opts)
-
-  if '--test' in opts:
+  import argparse, re, sys, textwrap
```
Comment (modocache) [inline src/codemod.py]: Would it be better to import these at the top level? I think that's what this PEP suggests: https://www.python.org/dev/peps/pep-0008/#imports

Also, I know this isn't consistent within codemod itself, so I'm sorry to point it out here. But I do plan on making this file a little more PEP compliant, and it wouldn't hurt to start now! :speak_no_evil:

**Comment by modocache** [comment]
This is *awesome*--I just had one small nitpick, the comment I made inline. Thanks for the contribution, and sorry it took so long to review! :100: :sparkling_heart:

**Comment by modocache** [comment]
Thanks, @DanNixon! :+1:

**Comment by DanNixon** [comment]
No problem.

**Pull request merged**